homeassistant/components/ecobee/binary_sensor.py

```diff
@@ -67,14 +67,7 @@ def device_info(self):
                         f"{ECOBEE_MODEL_TO_NAME[thermostat['modelNumber']]} Thermostat"
                     )
                 except KeyError:
-                    _LOGGER.error(
-                        "Model number for ecobee thermostat %s not recognized. "
-                        "Please visit this link and provide the following information: "
-                        "https://github.com/home-assistant/core/issues/27172 "
-                        "Unrecognized model number: %s",
-                        thermostat["name"],
-                        thermostat["modelNumber"],
-                    )
+                    model = "Unknown Thermostat"
```

Comment:
If we don't know the model we set it to `None` or don't include it in the device info at all.

Reply:
Updated to set the model to `None` if unknown.